File: backend/jufa-backend/src/main/java/ml/jufa/backend/b2b/entity/ProductUnit.java
package ml.jufa.backend.b2b.entity;

public enum ProductUnit {
    PIECE("Pièce"),
    CARTON("Carton"),
    PACK("Pack"),
    KG("Kilogramme"),
    LITRE("Litre"),
    SACK("Sac"),
    BOX("Boîte");

    private final String displayName;

    ProductUnit(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}

File: mobile/lib/features/b2b/domain/entities/product_unit.dart
enum ProductUnit {
  piece,
  carton,
  pack,
  kg,
  litre,
  sack,
  box;

  String get displayName {
    switch (this) {
      case ProductUnit.piece:
        return 'Pièce';
      case ProductUnit.carton:
        return 'Carton';
      case ProductUnit.pack:
        return 'Pack';
      case ProductUnit.kg:
        return 'Kilogramme';
      case ProductUnit.litre:
        return 'Litre';
      case ProductUnit.sack:
        return 'Sac';
      case ProductUnit.box:
        return 'Boîte';
    }
  }

  static ProductUnit fromString(String value) {
    return ProductUnit.values.firstWhere(
      (e) => e.name.toUpperCase() == value.toUpperCase(),
      orElse: () => ProductUnit.piece,
    );
  }
}
